content/en/references/configuration.md

```diff
@@ -308,40 +306,90 @@ To learn more about these configuration options, see [DNS Server]({{< ref "dns-s
+| `LAMBDA_STAY_OPEN_MODE` | 3.0.0 | `1` (default) | Usage of the stay-open mode of Lambda containers. Only applicable if `LAMBDA_EXECUTOR=docker-reuse`. Set to `0` if you want to use [Hot Reloading]({{< ref "hot-reloading" >}}).<br> **Removed in new provider because stay-open mode is the default behavior. `LAMBDA_KEEPALIVE_MS` can be used to configure how long containers should be kept running in-between invocations.** |
+| `LAMBDA_EXECUTOR` | 3.0.0 |  | Method to use for executing Lambda functions. For `docker` and `docker-reuse`, if LocalStack itself is started inside Docker, then the `docker` command needs to be available inside the container (usually requires to run the container in privileged mode). More information in Lambda Executor Modes.<br> **Removed in new provider. Mount the Docker socket or see [migration guide]({{< ref "user-guide/aws/lambda" >}}).** |
+| | | `docker` (default) | Run each function invocation in a separate Docker container. |
+| | | `local` (fallback) | Run Lambda functions in a temporary directory on the local machine. |
+| | | `docker-reuse` | Create one Docker container per function and reuse it across invocations. |
+| `LAMBDA_JAVA_OPTS` | 3.0.0 | `-Xmx512M` | Allow passing custom JVM options to Java Lambdas executed in Docker. Use `_debug_port_` placeholder to configure the debug port, e.g., `-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=_debug_port_`.<br> **Currently not supported in new provider but possible via custom entrypoint.** |
+| `PERSIST_ALL` | 3.0.0 | `true` (default) | Whether to persist all resources (including user code like Lambda functions), or only "light-weight" resources (e.g., SQS queues, or Cognito users). Can be set to `false` to reduce storage size of `DATA_DIR` folders or Cloud Pods. |
+| `KMS_PROVIDER` | 3.0.0 | `moto` (default), `local-kms` | `local-kms` has been removed. |
+| `KINESIS_PROVIDER` | 3.0.0 | `kinesis-mock` (default) and `kinesalite` | |
+| `EDGE_FORWARD_URL` | 3.0.0 | | Optional target URL to forward all edge requests to (e.g., for distributed deployments) |
+| `MOCK_UNIMPLEMENTED` | 3.0.0 | | Whether to return mocked success responses (instead of 501 errors) for currently unimplemented API methods |
+| `LEGACY_IAM_PROVIDER` | 0.0.0 | `0` (default)\|`1` | Enables the pre-1.0 legacy IAM provider |
+| `SQS_PROVIDER` | 0.0.0 |  `moto` (default) and `elasticmq` |
+| `S3_MOUNT` | 0.0.0 | | Configure a global parent directory that contains all buckets as sub-directories (`S3_MOUNT=/path/to/root`) or an individual directory that will get mounted as special bucket names (`S3_MOUNT=/path/to/root/bucket1:bucket1`) |
+| `KINESIS_INITIALIZE_STREAMS` | 0.0.0 | `"my-first-stream:1,my-other-stream:2:us-west-2,my-last-stream:1"` | A comma-delimited string of stream names, its corresponding shard count and an optional region to initialize during startup. If the region is not provided, the default region is used. Only works with the `kinesis-mock` `KINESIS_PROVIDER`. |
```

Comment:
FYI these are the ones where I wasn't sure yet when we first removed them. It's kinda hard to track down some of these configs

Reply:
One of them was resolved with the help of @bentsku  at least: `S3_MOUNT` funnily enough never actually existed except during the live of one PR in localstack :sweat_smile:

Reply:
The only one now missing is the `LEGACY_IAM_PROVIDER`. Had a hard time so far finding that in the code base at any point in time, maybe @dfangl has  some insights?

Reply:
As discussed in Slack, not aware of this variable. There was only `LEGACY_IAM_ENFORCEMENT` I am aware of.

Reply:
I also didn't find anything in the full git history `git rev-list --all | xargs git grep LEGACY_IAM_PROVIDER` (both localstack and ext)

Reply:
Yeah already removed that one, not sure where it came from